### PR TITLE
storage: Add CreateDir

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/beyondstorage/go-storage/issues/560

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -25,13 +25,38 @@ const (
...
 	o.ID = s.getAbsPath(path)
 	o.Path = path
 	return o
 }
 
+func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCreateDir) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+
+	err = s.bucket.Mkdir(rp)
+	if err != nil && checkErrorCode(err, 40600002) {
```
Comment (Xuanwo) [inline storage.go]: Maybe we need to extract `40600002` as const?

Comment (JinnyYi) [reply storage.go]: LGTM

File: utils.go [hunk 2]
```diff
@@ -104,10 +105,10 @@ func formatError(err error) error {
 		default:
 			return fmt.Errorf("%w, %v", services.ErrUnexpected, err)
 		}
-	case fn("40400001"):
+	case fn(strconv.Itoa(responseCodeFileOrDirectoryNotFound)):
```
Comment (Xuanwo) [inline utils.go]: Can we also can the code like we does in `checkErrorCode`

Comment (JinnyYi) [reply utils.go]: I‘ve updated as my understanding. Please help to review.

File: utils.go [hunk 3]
```diff
@@ -90,30 +91,24 @@ func formatError(err error) error {
...
-
-	switch {
-	case !fn(""):
-		// If body is empty
-		switch {
-		case strings.Contains(err.Error(), "404"):
+	if ae, ok := err.(*upyun.Error); ok {
+		switch ae.Code {
+		case responseCodeFileOrDirectoryNotFound, responseCodeNotFoundMarkAsDeleted, responseCodeNotFoundBlockDeleted:
+			// 40400001: file or directory not found
```
Comment (Xuanwo) [inline utils.go]: `40400001` has been extracted as const, we can add comment upon there.

File: storage.go [hunk 4]
```diff
@@ -25,13 +25,38 @@ const (
...
+	err = s.bucket.Mkdir(rp)
+	if err != nil && checkErrorCode(err, responseCodeFolderAlreadyExist) {
+		// Omit `folder already exists` error here.
+		err = nil
+	}
+	if err != nil {
+		return
+	}
+
+	o = s.newObject(true)
```
Comment (Xuanwo) [inline storage.go]: If the dir has been created, we need to use `s.newObject(false)` to allow the user to fetch data from the remote.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**